### PR TITLE
Backend validation

### PR DESCRIPTION
**Pull request by danielgrijalva**

### CODE REVIEW

File: backend/controllers/contact.js [hunk 1]
```diff
@@ -1,41 +1,51 @@
...
-   findAll,
-   create,
-   destroy,
-   update
- }
\ No newline at end of file
+const Contact = require("../models/contact");
+
+async function findAll(ctx) {
+  const contacts = await Contact.find({});
```
Comment (fhernandezn) [inline backend/controllers/contact.js]: Missing catch

File: backend/controllers/contact.js [hunk 2]
```diff
@@ -1,41 +1,51 @@
...
+}
+
+async function create(ctx) {
+  const newContact = new Contact(ctx.request.body);
+  try {
+    const savedContact = await newContact.save();
+    ctx.body = savedContact;
+  } catch (e) {
+    const errors = [];
+    for (var field in e.errors) {
```
Comment (fhernandezn) [inline backend/controllers/contact.js]: You can use a map function instead of this

File: backend/controllers/contact.js [hunk 3]
```diff
@@ -1,41 +1,51 @@
...
+    }
+
+    ctx.status = e.status || 500;
+    ctx.body = { errors: errors };
+  }
+}
+
+async function destroy(ctx) {
+  const id = ctx.params.id;
+  const contact = await Contact.findById(id);
```
Comment (fhernandezn) [inline backend/controllers/contact.js]: Missing catch here as well

File: backend/models/contact.js [hunk 4]
```diff
@@ -1,16 +1,63 @@
 const mongoose = require("mongoose");
+var uniqueValidator = require("mongoose-unique-validator");
```
Comment (fhernandezn) [inline backend/models/contact.js]: Why mixing var and const?

Comment (danielgrijalva) [reply backend/models/contact.js]: Ok, I'm gonna use const since `uniqueValidator` will not change

File: backend/controllers/contact.js (hunk 3 above)
Comment (danielgrijalva) [reply backend/controllers/contact.js]: Yes, maybe I should've named this pull "create contact validation". I'm gonna add more error handling to the other functions before merging!

File: backend/controllers/contact.js (hunk 2 above)
Comment (danielgrijalva) [reply backend/controllers/contact.js]: I tried but `e.errors` isn't an array

Comment (danielgrijalva) [reply backend/controllers/contact.js]: I was able to replicate it with:
```javascript
    Object.keys(e.errors).map(function(field) {
      errors.push(e.errors[field].message);
    });
```
But it doesn't seem very readable to me... does it improve performace?

**Pull request merged**